lib/improver/cli/weighted_blending.py

```diff
@@ -173,30 +172,127 @@ def main(argv=None):
-        model_id_attr=args.model_id_attr,
-        spatial_weights=args.spatial_weights_from_mask,
-        fuzzy_length=args.fuzzy_length)
+    # Load cubes to be blended.
+    cubelist = load_cubelist(args.input_filepaths)
+
+    result = process(cubelist, weights_dict, args.wts_calc_method,
+                     args.coordinate, args.cycletime, args.y0val, args.ynval,
+                     args.cval, args.weighting_coord, args.model_id_attr,
+                     args.spatial_weights_from_mask, args.fuzzy_length)
 
     save_netcdf(result, args.output_filepath)
 
 
+def process(cubelist, weights_dict, wts_calc_method, coordinate, cycletime,
+            y0val, ynval, cval, weighting_coord,
+            model_id_attr='mosg__model_configuration',
+            spatial_weights_from_mask=False, fuzzy_length=20000.0):
+    """Module to run weighted blending.
+
+    Load in arguments and ensure they are set correctly.
+    Then load in the data to blend and calculate default weights
+    using the method chosen before carrying out the blending.
+
+    Args:
+        cubelist (iris.cube.Cubelist):
+            Cubelist of cubes to be blended.
+        wts_calc_method (str):
+            Method to use to calculate weights used in blending.
+            "linear" (default): calculate linearly varying blending weights.
+            "nonlinear": calculate blending weights that decrease
+            exponentially with increasing blending coordinates.
+            "dicts": calculate weights using a dictionary passed in.
+        weights_dict (dictionary):
+            Dictionary from which to calculate blending weights. Dictionary
+            format is as specified in the
+            improver.blending.weights.ChoosingWeightsLinear
+        coordinate (str):
+            The coordinate over which the blending will be applied.
+        cycletime (str):
+            The forecast reference time to be used after blending has been
+            applied, in the format YYYYMMDDTHHMMZ. If not provided, the
+            blended file take the latest available forecast reference time
+            from the input cubes supplied.
+        y0val (float):
+            The relative value of the weighting start point (lowest value of
+            blend coord) for choosing default linear weights.
+            This must be a positive float or 0.
+        ynval (float):
+            The relative value of the weighting end point (highest value of
+            blend coord) for choosing default linear weights. This must be a
+            positive float or 0.
+            Note that if blending over forecast reference time, ynval >= y0val
+            would normally be expected (to give greater weight to the more
+            recent forecast).
+        cval (float):
+            Factor used to determine how skewed the non-linear weights will be.
+            A value of 2 implies equal weighting. If not set, a default value
+            of cval=0.85 is set in the plugin.
```

Comment:
This should have been changed in the CLI (I will bug fix it once you are finished). There is no default value for cval anymore, hence it is now a required arg rather than kwarg. Not your fault, but can you remove the stuff about the default being 0.85 here.